[docs] "What's new in Svelte" December newsletter (#8064)
sveltejs/svelte: site/content/blog/2022-12-01-whats-new-in-svelte-december-2022.md

---
title: "What's new in Svelte: December 2022"
description: "Rounding the corner to SvelteKit 1.0"
author: Daniel Sandoval
authorURL: https://desandoval.net
---

> SvelteKit V1 is just around the corner! With [99% of the 1.0 milestones completed](https://github.com/sveltejs/kit/milestone/2), there's a lot of new changes from the last month to cover...

Reviewer: ```suggestion
SvelteKit 1.0 is just around the corner! With [99% of the milestone issues completed](https://github.com/sveltejs/kit/milestone/2), there's a lot of new changes from the last month to cover...
```

Let's get to it!

## What's new in SvelteKit
- Use the `willUnload` property to find out if the navigation will result the app being unloaded (full page reload/closing/leaving to another page). ([#6813](https://github.com/sveltejs/kit/pull/6813))

> - `__data.json` requests now use the Vary header, which allows one cached response for each variation of the x-sveltekit-invalidated header, ensuring we cache matching responses for all invalidation scenarios ([#7532](https://github.com/sveltejs/kit/pull/7532))

Reviewer: ```suggestion
- `__data.json` requests now use the `Vary` header, which allows one cached response for each variation of the `x-sveltekit-invalidated` header, ensuring we cache matching responses for all invalidation scenarios ([#7532](https://github.com/sveltejs/kit/pull/7532))
```

---
title: "What's new in Svelte: December 2022"
description: "Rounding the corner to SvelteKit 1.0"
author: Daniel Sandoval
authorURL: https://desandoval.net
---

SvelteKit 1.0 is just around the corner! With [99% of the milestone issues completed](https://github.com/sveltejs/kit/milestone/2), there's a lot of new changes from the last month to cover...

Let's get to it!

## What's new in SvelteKit
- Use the `willUnload` property to find out if the navigation will result the app being unloaded (full page reload/closing/leaving to another page). ([#6813](https://github.com/sveltejs/kit/pull/6813))

> - `__data.json` requests now use the `Vary` header, which allows one cached response for each variation of the `x-sveltekit-invalidated` header, ensuring we cache matching responses for all invalidation scenarios ([#7532](https://github.com/sveltejs/kit/pull/7532))

Reviewer 2: We realized that not all platforms/CDNs support this header, so this will change - I'll reword it to something more generic

```suggestion
- `__data.json` requests now allows for caching while ensuring we cache matching responses for all invalidation scenarios ([#7532](https://github.com/sveltejs/kit/pull/7532))
```

- [Sky Cart: An Open Source, cloud-agnostic shopping cart using Stripe Checkout](https://dev.to/stripe/sky-cart-an-open-source-cloud-agnostic-shopping-cart-using-stripe-checkout-o5k) by Mike Bifulco for Stripe

**Libraries, Tools & Components**
- [Threlte](https://threlte.xyz/) is a component library for Svelte to build and render three.js scenes declaratively and state-driven in Svelte apps. It's being featured again to highlight the new "Playground" button in its examples
- [Svelte Turnstile](https://github.com/ghostdevv/svelte-turnstile) is a library to integrate Cloudflare's Turnstile (a new CAPTCHA alternative) into a Svelte app
- [ActionStore](https://github.com/buhrmi/actionstore) allows you to push data directly into Svelte stores via ActionCable
- [SvelteKit + `<is-land>`](https://sveltekit-is-land.vercel.app/) is an experiment with partial hydration in SvelteKit using `@11ty/is-land`
- [Svelte Color Select](https://github.com/CaptainCodeman/svelte-color-select) is an Okhsv Color Selection component for Svelte using OKLab perceptual colorspace

> - [svelte-awesome-color-picker](https://github.com/Ennoriel/svelte-awesome-color-picker) is a highly customizable color picker component library

Reviewer 3: ```markdown
[Svelte ColourPicker](https://github.com/cooolbros/svelte-colourpicker) is a lightweight opinionated color picker component
````
